atom/browser/api/atom_api_web_contents.cc

```diff
@@ -789,8 +790,14 @@ void WebContents::AddNewContents(content::WebContents* source,
     if (disposition != WindowOpenDisposition::NEW_WINDOW &&
         disposition != WindowOpenDisposition::NEW_POPUP) {
       if (!browser) {
-        // TODO(bridiver) - this should be the source owner_window
-        browser = owner_window()->browser();
+        brave::TabViewGuest* guest =
+              brave::TabViewGuest::FromWebContents(source);
+        CHECK(guest);
+        auto relay = atom::NativeWindowRelay::FromWebContents(
```

Comment:
this is actually problematic because it assumes the source is attached. It would be much simpler to get the browser from the TabHelper for the source anyway